### ISSUE TITLE
Dynamic Power Management?

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#618]
Originally reported of Google Code with ID 562.

```
Debian Sysadmins run several clusters with blade centers and shared storage. 

It would be nice, if we could shut down unneeded blades to save some power.

My current idea would be to have a daemon that uses hbal to achieve that.
One could drain a node and try to group as many instances together and drain the next node, until that fails.

As with shared storage you do not need to do storage migration, this only needs to migrate CPU instructions and the content of the RAM, thus it should not be that time consuming.

Ganeti already can (as far as i see) deal with the IPMI/ILO/RSA/KVM interfaces of several machines, so shutting down hardware or starting it up again should not be a big issue.
```
Originally added on 2013-09-04 09:31:14 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-09-04 10:08:33 +0000 UTC.

Added Labels: Type-Enhancement
Changed State: Accepted
Added to Milestone: Release2.11

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 5f6e9cb938cc2b3f35e48ea194b1d605a8c6efe3
Author: Klaus Aehlig <aehlig@google.com>
Date:   Thu Oct 24 11:24:07 2013 +0200

    Add hsqueeze planning for compression
    
    Add a new htool, hsqueeze, for dynamic power management. This
    commit only implements the first useful part: plan (but not
    executing) taking nodes offline nodes still keeping within the
    resource limit.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Michele Tartara <mtartara@google.com>


commit 268fb0bdb77bb3d36a725dbcc9afc11cba43fa18
Author: Klaus Aehlig <aehlig@google.com>
Date:   Thu Nov 14 15:29:20 2013 +0100

    hsqueeze: add option to show or save commands
    
    Add an option to hsqueeze to show, or save in a file, the commands
    that have to be carried out.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Hrvoje Ribicic <riba@google.com>

```
Originally added on 2013-11-15 15:44:57 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
2.12 release is already cut

```
Originally added on 2014-04-28 12:37:21 +0000 UTC.

Added to Milestone: Release2.13
Removed from Milestone: Release2.11

**Comment by Ganeti-Issues-Migrator** [comment]
```
hsqueeze does planning, however the -X option is still missing. Unassigning
to allow others to take over.
```
Originally added on 2014-05-09 10:22:01 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit c4027661842fc0904f6d7bca4c3c252db4e5a482
Author: BSRK Aditya <bsrk@google.com>
Date:   Fri Jul 18 04:46:30 2014 +0000

    Add -X option to hsqueeze's man page
    
    Final patch in the patch series.
    
    Signed-off-by: BSRK Aditya <bsrk@google.com>
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Klaus Aehlig <aehlig@google.com>

commit 7aae9a30b9069f6cd5202cd7ccaec199e1f1197b
Author: Bsrk Aditya <bsrk@google.com>
Date:   Fri Jul 18 04:46:29 2014 +0000

    Add execute option to Hsqueeze
    
    Generate opcodes and execute them using execWithCancel
    
    Signed-off-by: BSRK Aditya <bsrk@google.com>
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Klaus Aehlig <aehlig@google.com>

commit 09c94501aa628d050e4b0073bbfbdabd34b25335
Author: BSRK Aditya <bsrk@google.com>
Date:   Fri Jul 18 04:46:28 2014 +0000

    Add a fct to generate the OpCode for adding tags to a node
    
    Will be used by hsqueeze.
    
    Signed-off-by: BSRK Aditya <bsrk@google.com>
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Klaus Aehlig <aehlig@google.com>

commit f34e28defc186892f4a92d493696dad7d743a1b7
Author: BSRK Aditya <bsrk@google.com>
Date:   Fri Jul 18 04:46:27 2014 +0000

    Add a function to generate the opcodes to power off nodes
    
    Will be used by Hsqueeze to power off nodes
    
    Signed-off-by: BSRK Aditya <bsrk@google.com>
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Klaus Aehlig <aehlig@google.com>

commit 9b03c61d7961e1e6da1f16635e0a0e1c08b0758f
Author: BSRK Aditya <bsrk@google.com>
Date:   Fri Jul 18 04:46:26 2014 +0000

    Add a function to generate the opcode to power on nodes
    
    Will be used by hsqueeze to power on nodes.
    
    Signed-off-by: BSRK Aditya <bsrk@google.com>
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Klaus Aehlig <aehlig@google.com>

```
Originally added on 2014-09-12 10:54:50 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.14.0
```
Originally added on 2015-06-02 10:37:02 +0000 UTC.

Changed State: Released
